### ISSUE TITLE
Conciliate pyopy naming with hctsa

### ISSUE THREAD

**Issue by sdvillal** [strawlab/pyopy#2]
Properly defining semantics for the main abstractions of the library is important and, of course, it would make things easier to agree as much as possible with the way @benfulcher has given to hctsa.

Ben calls:
- "feature": a single number describing a time series.
- "operation": the mechanism by which a single feature is generated
- his conceptual mapping is 1 feature <-> 1 operation
- has not a good name for functions within .m files

He writes:

> I like that every feature has its own operation. Whether we add additional terms for methods that produce multiple outputs is up to the implementation. I guess we need a word for each .m file — I’ve called them code files or methods in the past.

My stand at the moment is:
- We agree for "feature" and "operation"
- I think the mapping 1 operation -> n features makes more sense; 
  - reallocation of feature, operation would still be perfectly possible
  - makes for a more natural mapping to software artifacts (whether .m files or python classes)
- We need to name software functions and function calls with concrete parameters.
  - functions can just be functions
  - function calls with concrete parameters could just still be "operations"
  - if overloading operation to also mean "a function call with concrete parameters" is too misleading whenever talking with Ben, we can alternatively use something fun like _"curry"_ (honoring [partial application](http://en.wikipedia.org/wiki/Partial_application) and [currying](http://en.wikipedia.org/wiki/Currying)).

To summarise, I would just use "feature", "operator" and "function". Once decided, will need to change the name of the [HCTSAFunctionCategory](https://github.com/strawlab/pyopy/blob/1f7b3283e716bba5db9e93fe0fdb231839c78fe6/pyopy/hctsa/hctsa_catalog.py#L101) accordingly.
